src/offset/local/tz_info/timezone.rs

```diff
@@ -48,15 +46,21 @@ impl TimeZone {
         // attributes are not allowed on if blocks in Rust 1.38
         #[cfg(target_os = "android")]
         {
-            if let Ok(bytes) = android_tzdata::find_tz_data(tz_string) {
+            if let Ok(Some(bytes)) =
+                crate::offset::local::tzdata::find_tz_data_android_from_fs(tz_string)
+            {
                 return Self::from_tz_data(&bytes);
             }
         }
 
         // ohos merge all file into tzdata since ver35
         #[cfg(target_env = "ohos")]
         {
-            return Self::from_tz_data(&find_ohos_tz_data(tz_string)?);
+            if let Ok(Some(bytes)) =
+                crate::offset::local::tzdata::find_tz_data_ohos_from_fs(tz_string)
+            {
+                return Self::from_tz_data(&bytes);
+            }
```

Comment:
If we're going to do this, let's merge make the definition of `find_tz_data_*_from_fs()` dependent on the `target_os` directly instead of having the duplication here. Let's call the function `tz_data::for_zone()`.

(I want consistent spelling of `tz_data`, no mix of `tzdata` and `tz_data`. `tz_data` seems more idiomatic to me but if there is external precedent for using `tzdata` consistently that's okay too.)

Reply:
Done